### PR TITLE
Test Python 3.13 in CI

### LINKED ISSUE #471: Test Python 3.13 in CI

### PR DESCRIPTION
**Pull request by sydney-runkle**
Closes #471

Thanks @Kludex for supporting the `rye` -> `uv` migration to prep for this :).

Currently blocked by https://github.com/huggingface/tokenizers/issues/1639, I've opened a PR to get this unblocked.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -46,7 +47,8 @@ dependencies = [
     "opentelemetry-instrumentation >= 0.41b0",
     "rich >= 13.4.2",
     "protobuf >= 4.23.4",
-    "typing-extensions >= 4.1.0",
+    'typing-extensions>=4.1.0; python_version < "3.13"',
+    'typing-extensions>=4.12.2; python_version >= "3.13"',
```
Comment (Kludex) [inline pyproject.toml:51]: Why is this needed?

Comment (sydney-runkle) [reply pyproject.toml:51]: Becase `typing-extensions` < 4.12.2 doesn't support 3.13

Comment (sydney-runkle) [reply pyproject.toml:51]: Maybe that's not true, hmm. Looking now.

**Comment by alexmojaki** [comment]
It looks like both the core OTEL SDK and instrumentations are still working on 3.13 support: https://github.com/open-telemetry/opentelemetry-python/issues/4211

So let's deprioritize this and wait a bit. When I opened the issue it hadn't occurred to me that dependencies might be this much of a problem.

**Comment by sydney-runkle** [comment]
Sounds good, will close for now and check out some of those other issues you mentioned :).
